Add JRuby extension to the gem

headius opened the pull request:
This pulls in the nkf extension implementation from JRuby. The build and load logic has been updated along the same lines as ruby/digest and the gem appears to build correctly for the -java platform.

Fixes #13

headius committed:
Add JRuby extension to the gem

This pulls in the nkf extension implementation from JRuby. The
build and load logic has been updated along the same lines as
ruby/digest and the gem appears to build correctly for the -java
platform.

Fixes #13

headius commented:
@hsbt @mrkn I believe I have made the same changes here as in digest, other than adding JRuby to tests (I am not sure we pass them all and run a very old version in our repository).

headius committed:
Add JRuby to CI

mrkn commented:
I think you need to confirm @nurse's opinion about this.

nurse commented:
> I think you need to confirm @nurse's opinion about this.

Thank you for mentioning me. I commented that I agree to merge an implementation for JRuby somewhere.

headius committed:
Fix compile errors and deprecations against JRuby 9.5

The pack deprecation does not have a suitable replacement, so it
will need some changes in JRuby 9.6(?).

headius commented:
> I think you need to confirm @nurse's opinion about this.

Oops, sorry. Github suggested you but I should have checked actual maintainers.

headius commented:
Tests are running and green.

The only tricky part with these -java gems is that we really need them both to be pushed every time. The JRuby team is happy to maintain the Java code and help with releases. I'm not sure if @nobu or @hsbt have suggestions for how to make sure we always push both (other stdlib gems are doing this too).

hsbt commented on ext/java/org/jruby/ext/nkf/Command.java:
```diff
@@ -0,0 +1,59 @@
+/***** BEGIN LICENSE BLOCK *****
+ * Version: EPL 2.0/GPL 2.0/LGPL 2.1
```
@headius Should we handle `GPL 2.0` for JRuby version at https://github.com/ruby/nkf/pull/15/files#diff-981027bf28c71984fb019c36c09b0d12b2c84c185d306c6caede7ccf7a76d4a3R21 ?

headius replied:
We can just drop GPL altogether. EPL (business-friendly) + LGPL (GPL-compatible) should cover every case we need. Would that be ok?

hsbt replied:
(I'm not lawyer) EPL and LGPL are ok to me. Can you handle them under https://github.com/ruby/nkf/pull/15/files#diff-981027bf28c71984fb019c36c09b0d12b2c84c185d306c6caede7ccf7a76d4a3R34?

headius replied:
I have removed GPL (we know it scares people and have considered removing it from JRuby too) and added the additional JRuby licenses to the java gem.

headius committed:
Drop GPL and add licenses to gemspec

hsbt commented on nkf.gemspec:
```diff
@@ -28,8 +28,16 @@ Gem::Specification.new do |spec|
+
+  if Gem::Platform === spec.platform and spec.platform =~ 'java' or RUBY_ENGINE == 'jruby'
+    spec.platform = 'java'
+    spec.licenses      = ["EPL-2.0", "LGPL-2.1"]
```
```suggestion
    spec.licenses      += ["EPL-2.0", "LGPL-2.1"]
```

Sorry to nit-picking. gemspec or some files are distributed by Ruby or BSD-2-Clause.

headius replied:
Oops, that was supposed to be +=. Will fix.

headius replied:
Fixed in latest push.

headius commented:
I think everything on my end is ready. @hsbt @nurse Anything else needed?

hsbt commented:
@headius I'll cut of new version from master branch today.

flavorjones commented:
Thank you all for doing this work! :heart:
